src/elements/content-sidebar/activity-feed/Avatar.js

```diff
@@ -14,46 +14,48 @@ type Props = {
 
 type State = {
     avatarUrl?: ?string,
+    isPending: boolean,
 };
 
+// Note: setting a key is required, re-render with a new user is not implemented
+
 class Avatar extends React.PureComponent<Props, State> {
-    state = {
-        avatarUrl: null,
-    };
-
-    /**
-     * Success handler for getting avatar url
-     *
-     * @param {string} avatarUrl the user avatar url
-     */
-    getAvatarUrlHandler = (avatarUrl: ?string) => {
-        this.setState({
+    constructor(props: Props) {
+        super(props);
+        // short-circuit promises by checking if we already have a url
+        // or if we don't have a method to get one
+        const avatarUrl = props.user.avatar_url || null;
```

Comment:
Seems like this should be destructured or use `lodash/get`.

Reply:
`get` isn't typesafe, so I've been avoiding it. Destructuring looks like this:

```js
const { user: { avatar_url: avatarUrl = null } } = props
```

Reply:
Whats wrong with const avatarUrl: string = get(props, 'user.avatar_url');